app/assets/js/main.js

```diff
@@ -176,15 +176,40 @@ $('#check-hours').click(function() {
 
 	checkExists(studentid, function(exists) {
 		if (exists) {
-			getTime(studentid, function(time) {
-				swal({
-					type: 'info',
-					//title: 'Records Found',
-					text: 'You have spent ' + time.hours + ':' + time.minutes + ':' + time.seconds + ' in the shop.',
-					timer: 5000
-				}).then(focusInput, focusInput);
-
-				clearSubmit();
+			checkState(studentid, function(signedOut) {
+				if (signedOut) {
```

Comment:
The point here is that if you're signed out we don't update the db, but if you're signed in, we pretend like you pushed the sign out button and then we sign you in again, thus saving the person's hours.